### DISCUSSION TITLE
General question for automatic shutdown // mobile BirdNET-Pi

### DISCUSSION THREAD

**Discussion by MueJosh** [mcguirepr89/BirdNET-Pi#1136] [General]
I'm currently designing a fully mobile BirdNET and cost effective system. 
When it's finished and tested I'll write a little plan how I did it. 

The raspberry will be powered by a 1S 40P Powerbank. (3,7V /4,2) 
Output 5V and 9V (I only use 5V)

I want the system to be reliable, so no dead SD-Cards or burning lithium cells. 

Does someone know of a way to shutdown the pi if the voltage gets to 3,6V? 

I found the INA-219 Board that reads battery voltage, current and stuff- it's a small shunt with IC. 
I have no coding skills (scripting my way through the world), so I have no idea how I get the reading to shutdown the pi. 

I also thought about a dual relais system that forces a shutdown via gpio and a second one to power off the pi entirely (Timer Relais) but I couldn't find a IC (or circuit- I'm able to solder but would like to keep it simple) that reads voltage (e.g. 3,6V) and outputs a signal like a switch. 
This would be the safest way because the 02W draws 10ma even when it's off. 

The system doesn't need to turn on again, if the powerbank is empty someone has to change it with a full one. 
In the future I'll add a solar panel so it runs longer, but I still need a safety measurement for the pi. 

I don't consider a UPS for 30+€ because it still needs to be cost effective! And having two battery systems is not that great..

If someone has some code for an IC or a plan to shutdown automatically please let me know. 
Maybe Cron and read the voltage from the ina219? 

Thanks!

**Comment by pddpauw** [comment]
curious to read about your project, please keep me posted.  I've built several stand alone mobile units with 4G connection and VPN connection to allow accessing the webserver (you can find on my wiki).  All is powered with sun and battery, but not that cost effective...

First things that comes to my mind, is measuring voltage on one of the pins. If you go below the 1.8v (voltage divider network with couple of resistors) it is seen as logic 0, and you could use this to turn the Pi safely off (you can make a Python script to check this and send shutdown command).  Yet, those resistors will draw a (little) bit off current, but they are very cost efficient.

**Reply by MueJosh** [reply]
I'll definitely write a Summary when it's finished and working.
I also thought about the lte connection but in my areas of interest there is no or really slow internet. And in want to safe energy so the Pi runs longer.

I'll have a look at your wiki:)

I just have seen your answer, yesterday I already orderd some tl431 that does exactly the same with some resistors.
[https://community.element14.com/technologies/open-source-hardware/b/blog/posts/low-battery-warning-indicator]()
Thank you!

**Comment by Vollpflock** [comment]
Interesting. Which exact model of power bank are you using? I've tested several but haven't found a cost-effective option with substantial capacity yet.

**Reply by MueJosh** [reply]
> Interesting. Which exact model of power bank are you using? I've tested several but haven't found a cost-effective option with substantial capacity yet.

I go the diy route with 18650 cells and a powerbank module for 1S.
I have literally hundreds of them laying around 

If you go this way you can find such modules on Ali (for cheap), the cells can be ordered from nkon (currently there are some 21700, 4850mah cells for 0,79€; cheaper is not possible) and some 18650/21700 (whatever cell you have) brackets/holder. 

3S or 4S is safer for the cells but I don't want to step down from 12V because it's not that efficient. (As going from 3.7 to 5V) 



The low voltage shutdown problem is solved (untested but in theory it works) 
[https://community.element14.com/technologies/open-source-hardware/b/blog/posts/low-battery-warning-indicator]()
With a small relay it should safely shutdown.

**Comment by Vollpflock** [comment]
Thank you, this is a fantastic suggestion; I'll give it a try. Could you share which module you purchased on AliExpress? I'm an absolute beginner and unsure of what to look out for.

**Reply by MueJosh** [reply]
If you are new to this be careful with lithium! 
They can and will go up in flames if handled incorrectly. 

The simple way: 
1S20P for 18650 
[](https://www.aliexpress.com/item/1005006267361573.html?spm=a2g0o.home.pcJustForYouTop.1.22e712e2GjWkNf&gps-id=pcJustForYouTop&scm=1007.13562.333647.0&scm_id=1007.13562.333647.0&scm-url=1007.13562.333647.0&pvid=326ba70f-4619-428e-983d-021459e50431&_t=gps-id%3ApcJustForYouTop%2Cscm-url%3A1007.13562.333647.0%2Cpvid%3A326ba70f-4619-428e-983d-021459e50431%2Ctpp_buckets%3A668%232846%238111%231996&pdp_npi=4%40dis%21EUR%217.74%214.02%21%21%218.18%214.25%21%4021038e7717087686898551957e317e%2112000036535513360%21rec%21DE%213313670745%21&utparam-url=scene%3ApcJustForYouTop%7Cquery_from%3A)


The "hard" diy way: 
You could choose one without a display so it draws less current
[](https://www.aliexpress.com/item/1005006147441573.html?spm=a2g0o.order_list.order_list_main.16.47485c5fMogvhi&gatewayAdapt=glo2deu)
You have to solder this module to the terminals.
Don't solder lithium cells, if they overheat the go up in flames or get damaged. Spot welding is the way or you choose something like this: 
[](https://www.aliexpress.com/item/1005006416585140.html?spm=a2g0o.order_list.order_list_main.16.27001802k9LNj5)

Or 21700
[](https://www.aliexpress.com/item/1005004852731607.html?spm=a2g0o.productlist.main.1.92dc2438Aycmpe&algo_pvid=eba4a538-b179-4162-b42c-df07496586be&algo_exp_id=eba4a538-b179-4162-b42c-df07496586be-0&pdp_npi=4%40dis%21EUR%212.15%212.15%21%21%212.27%212.27%21%402103871e17087696420755606e8439%2112000030749726738%21sea%21DE%213313670745%21&curPageLogUid=aXUqFxssCjp8&utparam-url=scene%3Asearch%7Cquery_from%3A)


Just keep in mind that it takes some time until the powerbank is fully charged...

**Reply by MueJosh** [reply]
My links got removed.. 
With this information you'll find it :)

The easy way
![image](https://github.com/mcguirepr89/BirdNET-Pi/assets/145150281/fe24af10-da97-4e08-807c-49b41287204b)

The hard way
![image](https://github.com/mcguirepr89/BirdNET-Pi/assets/145150281/a39bb5c6-767e-45bb-8ea9-7700530c9132)


![image](https://github.com/mcguirepr89/BirdNET-Pi/assets/145150281/f47a936f-0b25-49e8-af65-20827e3d63bf)
![image](https://github.com/mcguirepr89/BirdNET-Pi/assets/145150281/e3cc4cf1-6129-4670-a996-f1543fb9cb17)

**Comment by Vollpflock** [comment]
Thank you very much! Regarding the easy way: 

20 x 18650 seems to be the almost the absolute maximum, isn't it? I couldn't find more than 21 x 18650 or 8 x 21700.

**Reply by MueJosh** [reply]
Yes, for plug and play 20/21 cells is the maximum. 
2.000mAh per cells  (or 8 5000mAh cells) = 40Ah 
(with the pi 02w this would give you around 5 days of runtime)
Charging the Powerbank:
1A charging ≈ 40h charging time 
2A charging ≈ 20h 
3A charging ≈ 14h
...

These boards are not designed for so much Ah.

( If you want more juice/ cells and don't want to wait so long until the BatteryBank/ Powerbank is fully charged you have to add cells in series so it can be charged with more voltage/ A. This adds complexity to the system and also makes it more expensive. (Different modules, more protection circuits and less efficiency if you step down to 5V) 
DiY With a setup like this isn't that great- LiFePo4 is much easier and safer - or back to car battery's.. 
)
After all, systems like the AudioMoth or BirdWeather Puc exist with a much longer runtime and in the end it's probably cheaper.

**Reply by Vollpflock** [reply]
Thank you very much and sorry for the late reply. Will buy a bunch of these

**Reply by MueJosh** [reply]
No problem! (Kein Ding ;) 

Please update me / us how it works out for you. 

Edit: did you order the 21700 cells?

**Reply by Vollpflock** [reply]
haven’t found big cases for 21700, so i bought cases for 18650. will slaughter my grandmothers ebike battery for that.

**Reply by MueJosh** [reply]
That should work too :D I also do it like this 
If you have any question, let me know.
